src/mlpack/methods/neighbor_search/neighbor_search_impl.hpp

```diff
@@ -274,16 +267,20 @@ NeighborSearch<SortPolicy,
     return *this; // Nothing to do.
 
   // Clean memory first.
-  if (treeOwner && referenceTree)
+  if (referenceTree)
+  {
     delete referenceTree;
-  if (setOwner && referenceSet)
+    referenceTree = NULL;
+  }
+  else
+  {
     delete referenceSet;
+    referenceSet = NULL;
```

Comment:
Technically setting the pointers to NULL isn't necessary anywhere in the code since either the object is being destructed or the pointers will then be set to something else, but it's not necessarily a problem to leave these lines in.

Reply:
You're right. I had thought the same initially, but then `oldFronNewReferences` wasn't getting cleared correctly in the `Train` method when switching from non-naive to naive mode. Specifically, the `MoveTrainTest` test was failing. Maybe, I can get rid of setting pointers to NULL and still have everything work correctly. I was thinking that it might be a good practice to set deleted pointers to NULL to avoid double delete crashes (but, again, my understanding of these topics very basic). Do you want me to give it one more go?

Reply:
Nah, I'll just remove the extra lines during merge. :)